src/code42cli/cmds/auditlogs.py

```diff
@@ -0,0 +1,185 @@
+    f = begin_option(
+        f,
+        AUDIT_LOGS_KEYWORD,
+        callback=lambda ctx, param, arg: parse_min_timestamp(arg),
+        required=True,
+    )
+    f = end_option(
+        f, AUDIT_LOGS_KEYWORD, callback=lambda ctx, param, arg: parse_max_timestamp(arg)
+    )
+    f = filter_option_event_types(f)
+    f = filter_option_usernames(f)
+    f = filter_option_user_ids(f)
+    f = filter_option_user_ip_addresses(f)
+    f = filter_option_affected_user_ids(f)
+    f = filter_option_affected_usernames(f)
+    return f
+
+
+@click.group(cls=OrderedGroup)
+@sdk_options(hidden=True)
+def audit_logs(state):
+    """Retrieve audit logs."""
+    pass
+
+
+@audit_logs.command()
+@filter_options
+@format_option
+@sdk_options()
+def search(
+    state,
+    begin,
+    end,
+    event_type,
+    username,
+    user_id,
+    user_ip,
+    affected_user_id,
+    affected_username,
+    format,
+):
+    _search(
+        state.sdk,
+        format,
+        begin_time=begin,
+        end_time=end,
+        event_types=event_type,
+        usernames=username,
+        user_ids=user_id,
+        user_ip_addresses=user_ip,
+        affected_user_ids=affected_user_id,
+        affected_usernames=affected_username,
+    )
+
+
+@audit_logs.command()
+@filter_options
+@server_options
+@send_to_format_options
+@sdk_options()
+def send_to(
+    state,
+    hostname,
+    protocol,
+    format,
+    begin,
+    end,
+    event_type,
+    username,
+    user_id,
+    user_ip,
+    affected_user_id,
+    affected_username,
+):
+    _send_to(
+        state.sdk,
+        hostname,
+        protocol,
+        format,
+        begin_time=begin,
+        end_time=end,
+        event_types=event_type,
+        usernames=username,
+        user_ids=user_id,
+        user_ip_addresses=user_ip,
+        affected_user_ids=affected_user_id,
+        affected_usernames=affected_username,
+    )
+
+
+def _search(sdk, format, **filter_args):
+
+    formatter = OutputFormatter(format, AUDIT_LOGS_DEFAULT_HEADER)
+    response_gen = sdk.auditlogs.get_all(**filter_args)
+
+    events = []
+    try:
+        for response in response_gen:
+            response_dict = json.loads(response.text)
+            if EVENT_KEY in response_dict:
+                events.extend(response_dict.get(EVENT_KEY))
+    except KeyError:
+        # API endpoint (get_page) returns a response without events key when no records are found
+        # e.g {"paginationRangeStartIndex": 10000, "paginationRangeEndIndex": 10000, "totalResultCount": 1593}
+        pass
+
+    event_count = len(events)
+    if not event_count:
+        click.echo("No results found.")
+    elif event_count > 10:
+        click.echo_via_pager(formatter.get_formatted_output(events))
+    else:
+        formatter.echo_formatted_list(events)
+
+
+def _send_to(sdk, hostname, protocol, format, **filter_args):
+    logger = get_logger_for_server(hostname, protocol, format)
+    response_handler = create_simple_send_to_handler(
```

Comment:
I don't really see the benefit of pulling this out into a separate function (`create_simple_send_to_handler`), at least not yet since there's only one place this is used. And there's no need to load the response.text to json as py42 already has already done it in the `Py42Response` object. Unless there's something I'm missing `_send_to()` could be simplified to: 

```python
def _send_to(sdk, hostname, protocol, format, **filter_args):
    logger = get_logger_for_server(hostname, protocol, format)
    with warn_interrupt():
        response_gen = sdk.auditlogs.get_all(**filter_args)
        for response in response_gen:
            if EVENT_KEY in response:
                for event in response[EVENT_KEY]:
                    logger.info(event)
            else:
                click.echo("No results found.")
```

Reply:
I intended to be consistent here with all "extraction/send-to-server" functions in a single module. Refactored, as requested. Are we fine here with `warn_interrupt` not having a message?

Reply:
Yeah, but this function doesn't actually use the extractor, so it doesn't quite fit there. 

I'd just rather not try to force abstracting out a function like this until we're sure we'll need that exact functionality in multiple places. 

And warn_interrupt does have a default message (`Cancelling operation cleanly, one moment...`) if none is passed in.